### PR TITLE
[docs] - Move assets into intro tutorial + up in sidenav [CON-18]

### PR DESCRIPTION
**Pull request by erinkcochran87**
### Summary & Motivation

This PR does two things:

1. Makes the assets guide into the primary **Intro tutorial**. Adds redirects and updates links to new URL where needed. [Preview](https://dagster-8sp6iyv0p-elementl.vercel.app/master/tutorial)
3. Moves the **Concepts > SDA** section up in the sidenav and re-orders the **Concepts** page to reflect the new order. [Preview](https://dagster-8sp6iyv0p-elementl.vercel.app/master/concepts)

### CODE REVIEW

File: docs/content/tutorial.mdx [hunk 1]
```diff
@@ -1,28 +1,45 @@
 ---
```
Comment (erinkcochran87) [inline docs/content/tutorial.mdx:1]: This page is one of the big changes.

File: docs/content/_navigation.json [hunk 2]
```diff
@@ -23,28 +23,24 @@
     ]
```
Comment (erinkcochran87) [inline docs/content/_navigation.json:23]: The sidenav updates are the other big part of the changes in this PR.

**Review by sryza — COMMENTED** [review]
The side nav changes for the concept pages look great.

I think the tutorial section could do a better job at guiding users and helping them understand how the concepts we're aiming to teach them about fit together. I'm concerned that a reader would come to it and not understand when the asset tutorial is relevant and when the ops & jobs tutorial is relevant.

Here are some ideas:
* Have wording at the top of the tutorials page saying something like "The main use of Dagster is to build and maintain data assets. The assets tutorial introduces you to the basic concepts for doing that. You should probably start here.  The asset functionality also sits on top of a more general orchestration engine that can be used for other stuff. This other op-focused tutorial explains how to use that."
* Change the op tutorial to focus on a use case that SDAs wouldn't be a good fit for (probably couldn't finish that in time for 0.15.0, but maybe we could do it in the next week or two?)
* Get rid of the Advanced Tutorials, because they're op focused and mostly don't fit our recommendations for best practices in the SDA-focused world.

Thoughts?

File: docs/content/tutorial.mdx [hunk 3]
```diff
@@ -1,28 +1,45 @@
...
-## Intro Tutorial
+---
+
+## Intro tutorials
+
+Install Dagster and master the basics with these tutorials.
+
+Before you get started, make sure you [**install Dagster**](/tutorial/intro-tutorial/setup).
+
+### Asset tutorials
```
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/content/tutorial.mdx:18]: I'm definitely not tied to the heading here. I was just trying to break up the content for these two concepts. Totally cool with going with something else.

**Comment by erinkcochran87** [comment]
> I think the tutorial section could do a better job at guiding users and helping them understand how the concepts we're aiming to teach them about fit together. I'm concerned that a reader would come to it and not understand when the asset tutorial is relevant and when the ops & jobs tutorial is relevant.

I agree completely. I got stuck and wanted to surface this for feedback instead of continuing to spin - this helps a lot!

> Have wording at the top of the tutorials page saying something like "The main use of Dagster is to build and maintain data assets. The assets tutorial introduces you to the basic concepts for doing that. You should probably start here. The asset functionality also sits on top of a more general orchestration engine that can be used for other stuff. This other op-focused tutorial explains how to use that."

I think this could solve the 'I don't know when this applies' problem, for sure.

> Change the op tutorial to focus on a use case that SDAs wouldn't be a good fit for (probably couldn't finish that in time for 0.15.0, but maybe we could do it in the next week or two?)

I think that's fine, but it definitely won't be in time for 0.15.0 😅 

> Get rid of the Advanced Tutorials, because they're op focused and mostly don't fit our recommendations for best practices in the SDA-focused world.

As in get rid of these **entirely**, remove them totally from the site?

**Comment by shalabhc** [comment]
It's great to have the "copy code" icon on every code sample. It would also be great to know what file name to save it as. Following the tutorial here: https://dagster-8sp6iyv0p-elementl.vercel.app/master/tutorial/intro-tutorial/assets/asset-graph#lets-get-serial, I can copy the code but I have to read ahead and infer the file name from the`dagit -f serial_asset_graph.py` example. Maybe we can show the filename in a title bar or caption on the code section itself? BTW, this is a problem with the currently live version as well.

**Comment by erinkcochran87** [comment]
I think that's a great suggestion @shalabhc! This won't fix the live version right now, but it will when we cut the new 0.15.0 release. I'll add it in.

File: docs/content/tutorial.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (sryza) [reply docs/content/tutorial.mdx]: Cool yeah thinking about it now, I guess I'm just advocating for removing the "s" at the end

**Comment by sryza** [comment]
@erinkcochran87 

> As in get rid of these entirely, remove them totally from the site?

I do mean that. There's definitely a cost to doing so, but here's why I think this probably makes sense:
* Most (probably all) of the content is also covered in the Concepts pages (although in a somewhat different style)
* The content does not represent best practices, so it's a little bit actively misleading

**Review by sryza — APPROVED** [review]
I like it!

**Pull request merged**